### ISSUE TITLE
no-extra-semi don't like block-less while statements

### ISSUE THREAD

**Issue by zyklus** [eslint/eslint#3817]
```js
while (!fn(--x));
```

`no-extra-semi` thinks that semi-colon is unnecessary :(

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by mysticatea** [comment]
@zyklus Thanks for the report.

What is the version of ESLint you are using?
We have a test case for the pattern since v1.0.0: https://github.com/eslint/eslint/blob/25f14ae9b4c862a81f336355a1029e02cb7f8fe0/tests/lib/rules/no-extra-semi.js#L25

**Comment by gyandeeps** [comment]
Please answer all the question as mentioned here https://github.com/eslint/eslint/issues/3817#issuecomment-140633787

**Comment by ilyavolodin** [comment]
Closing as there was no response to questions by the team. Feel free to reopen if you have new information.
